subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -114,16 +113,14 @@ void add_ipv6_addr_to_ot(struct openthread_context *context)
 
 	otIp6AddUnicastAddress(context->instance, &addr);
 
-#if CONFIG_OPENTHREAD_L2_LOG_LEVEL == SYS_LOG_LEVEL_DEBUG
-	{
+	if (CONFIG_OPENTHREAD_L2_LOG_LEVEL == LOG_LEVEL_DBG) {
```

Comment:
Why do you use run time if for this?

Reply:
Other option would have been to add `{ char buf[]; ... }` but that looked ugly.

Reply:
FWIW, this is not runtime in practice. It's the same thing as IS_ENABLED() (as far as I see), i.e. the branch should get stripped away from the final binary if the condition is not true

Reply:
Yes it should. Maybe not in -O0, but generally it should be stripped.